simon8491/src/main/java/workshop/UserDao.java

```diff
@@ -0,0 +1,124 @@
+    private static final String READ_USER_QUERY = """
+            SELECT * FROM users WHERE id = ?""";
+    private static final String UPDATE_USER_QUERY = """
+            UPDATE users SET username = ?, email = ?, password = ? WHERE id = ?""";
+    private static final String DELETE_USER_QUERY = """
+            DELETE FROM users WHERE id = ?""";
+    private static final String FIND_ALL_USERS_QUERY = """
+            SELECT * FROM users""";
+
+    public void createUser(User user) {
+        try (Connection connection = DbUtil.getConnection()) {
+            try (var stmtCreateUser = connection.prepareStatement
+                    (CREATE_USER_QUERY, PreparedStatement.RETURN_GENERATED_KEYS)) {
+                stmtCreateUser.setString(1, user.getUserName());
+                stmtCreateUser.setString(2, user.getEmail());
+                stmtCreateUser.setString(3, hashPassword(user.getPassword()));
+                stmtCreateUser.executeUpdate();
+                try (var rsCreateUser = stmtCreateUser.getGeneratedKeys()) {
+                    if (rsCreateUser.next()) {
+                        user.setId(rsCreateUser.getInt(1));
+                        System.out.println("Created user with ID: " + user.getId());
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            System.out.println("Problem with creating user in database:\n" + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+    }
+
+    public String hashPassword(String password) {
+        return BCrypt.hashpw(password, BCrypt.gensalt());
+    }
+
+    public User readUser(int id) {
+        try (Connection connection = DbUtil.getConnection()) {
+            try (var stmtReadUser = connection.prepareStatement(READ_USER_QUERY)) {
+                stmtReadUser.setInt(1, id);
+                try (var rsReadUser = stmtReadUser.executeQuery()) {
+                    if (rsReadUser.next()) {
+                        User user = new User();
+                        user.setId(rsReadUser.getInt("id"));
+                        user.setUserName(rsReadUser.getString("username"));
+                        user.setEmail(rsReadUser.getString("email"));
+                        //user.setPassword(rsReadUser.getString("password"));
+                        return user;
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            System.out.println("User cannot be read:\n" + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+        return null;
+    }
+
+    public void updateUser(User user) {
+        try (Connection connection = DbUtil.getConnection()) {
+            try (var stmtUpdateUser = connection.prepareStatement(UPDATE_USER_QUERY)) {
+                stmtUpdateUser.setString(1, user.getUserName());
+                stmtUpdateUser.setString(2, user.getEmail());
+                stmtUpdateUser.setString(3, this.hashPassword(user.getPassword()));
```

Comment:
BUG: Zde bych zrusil hashovani, protoze pokud uz updatujes zaheshovane heslo tak ho znova hashujes.

```suggestion
stmtUpdateUser.setString(3, user.getPassword());
```